lib/engine/game/base.rb

```diff
@@ -483,6 +484,15 @@ def sellable_turn?
         @turn > 1
       end
 
+      def route?(entity)
+        @graph.route?(entity)
+      end
+
+      def must_buy_train?(entity)
+        !entity.rusted_self && entity.trains.empty? &&
+        ((MUST_BUY_TRAIN == :route && route?(entity) == true) || MUST_BUY_TRAIN == :always)
```

Comment:
route?(entity) == true... tsk tsk

Reply:
why does route? return an optional?

Reply:
would mandatory be better?

Reply:
Some games (1882) have tiles which can be used on a route but do not count as a route for mandatory train buying